src/System.Net.Mime/tests/Unit/ByteEncodingTest.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+using Xunit;
+
+namespace System.Net.Mime.Tests
+{
+    public class ByteEncodingTest
+    {
+        [Theory]
+        [InlineData("some test header")]
+        [InlineData("some test header that is really long some test header that is really long some test header that is really long some test header that is really long some test header that is really long")]
+        public void EncodeHeader_WithNoUnicode_ShouldNotEncode(string testHeader)
+        {
+            string result = MimeBasePart.EncodeHeaderValue(testHeader, Encoding.UTF8, true);
+            Assert.False(result.StartsWith("=?utf-8?B?"));
+            Assert.False(result.EndsWith("?="));
```

Comment:
Nit: Should `StartsWith` and `EndsWith` be using `StringComparison.Ordinal` instead of current culture?